dolphin/drivers/dellemc_storage/__init__.py

```diff
@@ -0,0 +1,203 @@
+            self.password = access_info['password']
+        except KeyError as err:
+            LOG.error('ERROR: Invalid input register_info[' + str(err) + ']')
+            return storage
+
+        try:
+            # Initialise PyU4V connection to Unisphere
+            conn = PyU4V.U4VConn(
+                univmax_version=self.univmax_version,
+                server_ip=self.server_ip,
+                port=self.port,
+                verify=self.verify,
+                username=self.username,
+                password=self.password)
+
+            # Get the Unisphere version
+            version = conn.common.get_uni_version()
+
+            # Retrieve a list of arrays managed by your instance of Unisphere
+            array_list = conn.common.get_array_list()
+
+            # Close the session
+            conn.close_session()
+
+            if (version != self.univmax_version):
+                LOG.error('Storage version is different from supported version')
+
+            if (self.symmetrix_id not in array_list):
+                LOG.error('ERROR: Array list do not contain storage id')
+                return storage
+
+        except:
+            LOG.error('ERROR: Failed to connect to VMAX storage')
+            return storage
+
+        # Update storage_id's that this driver manages
+        if VMAXStorageDriver.storage_ids:
+            VMAXStorageDriver.storage_ids.append(self.storage_id)
+        else:
+            VMAXStorageDriver.storage_ids = [self.storage_id]
+        
+        storage = get_storage(self, context)
+
+        # Return all the storage ids this driver manage
+        LOG.info('register_storage(), driver successfully registered')
+        return storage
+
+    def get_storage(self, context):
+        Storage = {
+            'id': self.storage_id,
+            'name': 'VMAX',
+            'vendor': 'Dell EMC',
+            'description': '',
+            'model': '',
+            'status': '',
+            'serial_number': self.symmetrix_id,
+            'location': '',
+            'total_capacity': 0,
+            'used_capacity': 0,
+            'free_capacity': 0
+        }
+
+        try:
+            # Initialise PyU4V connection to Unisphere
+            conn = PyU4V.U4VConn(
+                univmax_version=self.univmax_version,
+                server_ip=self.server_ip,
+                port=self.port,
+                verify=self.verify,
+                username=self.username,
+                password=self.password)
+
+            LOG.info('Successfully connected storage')
+            Storage.status = 'Available'
+
+            # Get the Unisphere model
+            uri = '/system/symmetrix/' + self.symmetrix_id
+            model = conn.common.get_request(uri, '')
+            Storage.model = model['symmetrix'][0]['model']
+            LOG.info('Successfully retrieved storage model')
+
+            # Get first SRP
+            uri = '/sloprovisioning/symmetrix/' + self.symmetrix_id + '/srp'
```

Comment:
Will this code be re-usable for getting storage resource pool info too? Then can we move it to function.
Also, does SRP defines the Capacity of the storage? Please confirm. Just ensure that we don't need to care about the capacities of the Physical disks

Reply:
Above values are from single SRP that is thin provisioned. This will be updated.